src/Analyzers/MSTest.Analyzers/DataRowShouldBeValidAnalyzer.cs

```diff
@@ -205,16 +206,34 @@ private static void AnalyzeAttribute(SymbolAnalysisContext context, AttributeDat
 
             if (argumentType is not null && !argumentType.IsAssignableTo(paramType, context.Compilation))
             {
-                typeMismatchIndices.Add((currentArgumentIndex, Math.Min(currentArgumentIndex, methodSymbol.Parameters.Length - 1)));
+                int parameterIndex = Math.Min(currentArgumentIndex, methodSymbol.Parameters.Length - 1);
+                string parameterName = methodSymbol.Parameters[parameterIndex].Name;
+                string expectedType = paramType.ToDisplayString();
+                string actualType = argumentType.ToDisplayString();
+                typeMismatches.Add((parameterName, expectedType, actualType));
             }
         }
 
         // Report diagnostics if there's any type mismatch.
-        if (typeMismatchIndices.Count > 0)
+        if (typeMismatches.Count > 0)
         {
+            // Format all mismatches into a single message
+            string mismatchMessage;
+            if (typeMismatches.Count == 1)
+            {
+                var mismatch = typeMismatches[0];
+                mismatchMessage = $"Parameter '{mismatch.ParameterName}' expects type '{mismatch.ExpectedType}', but the provided value has type '{mismatch.ActualType}'";
```

Comment:
Please add this string in a resx (and update related xlf files). Also reuse the localized resource on line 230.